modules/ROOT/pages/apikit-workflow-convert-existing.adoc

```diff
@@ -1,104 +1,239 @@
-
-. Drag the following components from the Mule palette:
+* An *HTTP Listener* as a flow source
+* An *APIkit Router* operation within the flow process
 +
-* An HTTP Listener
-* An APIkit Router
+. In the properties editor for the *HTTP Listener*, in *Connector Configuration*, select the existing `HTTP_Listener_Config`.
+. Specify a listener path for the APIkit router. Enter `/api/*` in `Path` field.
+. In the properties editor for the APIkit router, in *Router Configuration*, click `Add` to create a new configuration.
 +
-. In the properties editor for APIkit Router, in Router Configuration, select `api-config`.
-. In the properties editor for HTTP Listener, in Connector Configuration, select api-httpListenerConfig, and set a path. For example, in Path, enter `/`.
-
-
-== Generate REST Backend Flows
-
-When you create a RAML in Studio, generating backend flows is optional. Error handling is generated for a RAML-based API. You need to reference generated flows from the main flow. This procedure assumes you have already created a main flow.
-
-To generate backend flows:
-
-. In the Package Explorer, right-click the project name.
-. Select Mule > Generate Flows From REST API.
-+
-The backend flows appear below the main flow.
-
-== Map RAML Resources to Actions
-
-As part of the backend-first design process, you map RAML resources to actions as shown in the following procedure. This procedure assumes you have generated backend flows.
-
-To map resources to actions:
-
-. Click the APIkit Router to open the *Properties Editor*. In *Router configuration*, click image:add-16x16.png["Add button."].
+The *Global Element Properties* wizard appears.
+. In *API Definition*, select the path `api/api.raml`.
+. Accept the configuration.
+. Delete the HTTP listener from `GET-Users-FLOW`.
+. Delete the HTTP listener from `GET-User-By-Id-FLOW`.
+
+[source,xml,linenums]
+----
+<mule>
+    <apikit:config outboundHeadersMapName="outboundHeadersMapName" httpStatusVarName="httpStatus" name="Router" api="api/api.raml" />
+    <http:listener-config name="HTTP_Listener_config">
+		<http:listener-connection host="0.0.0.0" port="8081" />
+	</http:listener-config>
+	<db:config name="Database_Config">
+		<db:my-sql-connection host="localhost" port="3306" user="admin" password="admin" database="usersdb" />
+	</db:config>
+    <flow name="apiFlow">
+		<http:listener config-ref="HTTP_Listener_config" path="/api/*"/>
+		<apikit:router config-ref="Router"/>
+	</flow>
+	<flow name="GET-Users-FLOW">
+		<db:select config-ref="Database_Config">
+			<db:sql ><![CDATA[SELECT * FROM USERS]]></db:sql>
+		</db:select>
+	</flow>
+	<flow name="GET-User-By-Id-FLOW">
+		<db:select config-ref="Database_Config" >
+			<db:sql ><![CDATA[SELECT * FROM USERS WHERE :userIdParameter == userID]]></db:sql>
+			<db:input-parameters ><![CDATA[#[{userIdParameter: attributes.queryParams.id}]]]></db:input-parameters>
+		</db:select>
+	</flow>
+</mule>
+----
+
+=== Step 3: Map API Resources-Actions to Flows
```

Comment:
Resources and Actions?

Reply:
Not sure. Requesting assistance from SMEs.

Reply:
Yes! "Action" makes reference to the "HTTP method", and to the HTTP method only. So I guess 'action' is not confusing since it is explained in the pattern and in the table below this title.